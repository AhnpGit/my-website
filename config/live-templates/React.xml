<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
