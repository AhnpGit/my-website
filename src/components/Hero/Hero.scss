.hero {
  color: $mischka;
  @include media-query($medium-up) {
    margin-top: $base-unit;
  }
}

.panel {
  display: flex;
  flex-direction: column;
}

.hero__background {
  position: relative;
  padding-bottom: 62%;
  background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.7)), url('https://drive.google.com/uc?id=0BwwULdUQlf8mdEx5ZUdyaDRZQ2s');
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-attachment: fixed;

  .hero__inner {
    display: flex;
    flex-direction: column;
    justify-content: center;
    position: absolute;
    width: 100%;
    height: 100%;

    @include media-query($medium-up) {
      padding: $base-unit;
    }

    h1 {
      text-transform: uppercase;
      font-weight: 300;
      margin-bottom: $base-unit;
    }

    .hero__links {
      display: flex;
    }
  }
}
