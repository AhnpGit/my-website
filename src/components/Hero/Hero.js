// @flow

import React from 'react'
import Button from '../../elements/Button/Button'

import './Hero.scss'
import Wrapper from '../../elements/Wrapper/Wrapper'

/**
 * Hero component displays the hero with image background and 2 buttons
 */
const Hero = () => (
  <section className="hero">
    <Wrapper>
      <div className="panel">
        <div className="hero__background">
          <div className="hero__inner">
            <h1>Goodbye junk food.<br />Hello super healthy meals.</h1>
            <div className="hero__links">
              <Button text="I'm hungry" onClick={() => {}} />
              <Button appearance="secondary" text="Show me more" onClick={() => {}} />
            </div>
          </div>
        </div>
      </div>
    </Wrapper>
  </section>
)

export default Hero
