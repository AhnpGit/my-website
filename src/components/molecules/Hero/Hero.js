// @flow

import React from 'react'
import Button from '../../atoms/Button/Button'

/**
 * Hero component displays the hero with image background and 2 buttons
 */
const Hero = () => (
  <section className="hero">
    <h1>Goodbye junk food.<br />Hello super healthy meals.</h1>
    <div className="hero__links">
      <Button text="I'm hungry" onClick={() => {}} />
      <Button appearance="secondary" text="Show me more" onClick={() => {}} />
    </div>
  </section>
)

export default Hero
