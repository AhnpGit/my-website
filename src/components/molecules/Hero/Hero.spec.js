import React from 'react'
import { shallow } from 'enzyme'
import Hero from './Hero'

describe('Hero component', () => {
  it('renders correctly without issues', () => {
    const result = shallow(
      <Hero />
    )

    expect(result).toMatchSnapshot()
  })
})
