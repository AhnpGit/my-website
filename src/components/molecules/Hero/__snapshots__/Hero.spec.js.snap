// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders correctly without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="hero"
>
    <h1>
        Goodbye junk food.
        <br />
        Hello super healthy meals.
    </h1>
    <div
        className="hero__links"
    >
        <Button
            onClick={[Function]}
            text="I'm hungry"
        />
        <Button
            appearance="secondary"
            onClick={[Function]}
            text="Show me more"
        />
    </div>
</section>,
  "nodes": Array [
    <section
      className="hero"
>
      <h1>
            Goodbye junk food.
            <br />
            Hello super healthy meals.
      </h1>
      <div
            className="hero__links"
      >
            <Button
                  onClick={[Function]}
                  text="I'm hungry"
            />
            <Button
                  appearance="secondary"
                  onClick={[Function]}
                  text="Show me more"
            />
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Hero />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="hero"
>
          <h1>
                    Goodbye junk food.
                    <br />
                    Hello super healthy meals.
          </h1>
          <div
                    className="hero__links"
          >
                    <Button
                              onClick={[Function]}
                              text="I'm hungry"
                    />
                    <Button
                              appearance="secondary"
                              onClick={[Function]}
                              text="Show me more"
                    />
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="hero"
>
          <h1>
                    Goodbye junk food.
                    <br />
                    Hello super healthy meals.
          </h1>
          <div
                    className="hero__links"
          >
                    <Button
                              onClick={[Function]}
                              text="I'm hungry"
                    />
                    <Button
                              appearance="secondary"
                              onClick={[Function]}
                              text="Show me more"
                    />
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Hero />,
}
`;
